src/main/java/lottery/domain/LotteryGame.java

```diff
@@ -13,12 +13,14 @@ public LotteryGame(int price, int pickCounts) {
     }
 
     public int buyNumberOfLotteryTickets(int cost) {
+        if (cost < price) {
+            throw new IllegalArgumentException("로또 1장의 가격은 " + this.price + "입니다.");
+        }
         return cost / this.price;
     }
 
-    public List<LotteryTicket> buyLotteryTickets(int numberOfTickets) {
-        AutoBuyBehavior autoBuyBehavior = new AutoBuyBehavior();
-        return autoBuyBehavior.buyLotteryTickets(this.pickCounts, numberOfTickets);
+    public List<LotteryTicket> buyLotteryTickets(int numberOfTickets, AutoBuyBehavior behavior) {
```

Comment:
```suggestion
    public List<LotteryTicket> buyLotteryTickets(int numberOfTickets, BuyBehavior behavior) {
```
구체적인 클래스 보다는 인터페이스로 선언한다면 더욱 유연하게 코드를 구현할수 있을거에요 😄

Reply:
네 해당거는 의견 주신대로 해당 방법이 맞는것 같습니다. 👍